### DISCUSSION TITLE
Is possible to put icon in front of ui.input bar?

### DISCUSSION THREAD

**Discussion by lkklkklkkj** [zauberzeug/nicegui#3980] [Q&A] [answered]
### Question

I want to put icon in front of input bar like the picture below

![image](https://github.com/user-attachments/assets/0d4dc3ca-d576-428d-a32e-1e7bb685d065)

The picture is from [Quasar QInput API](https://quasar.dev/vue-components/input).

**Comment by falkoschindler** [comment, marked as answer]
Hi @lkklkklkkj,

You can use the "prepend" slot like they do in the Quasar example:
```py
with ui.input(label='Label').add_slot('prepend'):
    ui.icon('event')
```
![Screenshot 2024-11-12 at 11 40 20](https://github.com/user-attachments/assets/fadc465d-598b-4912-be7a-373e5fb29bde)
